Add IDL skeleton for WebTransport

achristensen07 opened the pull request:
#### a4e9bc6fb756d57c2e3bad540d3330aa7d45fd19
<pre>
Add IDL skeleton for WebTransport
<a href="https://bugs.webkit.org/show_bug.cgi?id=260810">https://bugs.webkit.org/show_bug.cgi?id=260810</a>
rdar://114578275

Reviewed by Chris Dumez.

This adds IDLs based on the spec at <a href="https://www.w3.org/TR/webtransport/">https://www.w3.org/TR/webtransport/</a>
They are off by default and have simple stub implementations that don&apos;t do anything yet.
I also imported the web platform tests with a slight modification to webtransport/bidirectional-cancel-crash.https.html
to make it not time out when running with an incomplete implementation.

* LayoutTests/imported/w3c/web-platform-tests/webtransport/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/README.md: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection-ccns.https.tentative.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection-ccns.https.tentative.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection-ccns.https.tentative.window.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection.https.window.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection-ccns.https.tentative.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection-ccns.https.tentative.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection-ccns.https.tentative.window.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection.https.window.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/bidirectional-cancel-crash.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/bidirectional-cancel-crash.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.js: Added.
(const.url.of.BAD_URLS.test):
(const.options.of.OPTIONS.promise_test.async t):
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-fail.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-fail.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-fail.https.window.js: Added.
(set_csp):
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-pass.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-pass.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-pass.https.window.js: Added.
(set_csp):
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagram-cancel-crash.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagram-cancel-crash.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagram-cancel-crash.https.window.js: Added.
(test):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.js: Added.
(async write_datagrams):
(async write_N_datagrams):
(async read_datagrams):
(async write_numbers):
(async write_large_datagrams):
(async read_numbers_byob):
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/echo-large-bidirectional-streams.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/echo-large-bidirectional-streams.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/abort-stream-from-server.py: Added.
(session_established):
(stream_data_received):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/client-close.py: Added.
(session_established):
(stream_reset):
(stream_data_received):
(session_closed):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/custom-response.py: Added.
(connect_received):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/echo-request-headers.py: Added.
(session_established):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/echo.py: Added.
(session_established):
(stream_data_received):
(datagram_received):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/echo_datagram_length.py: Added.
(datagram_received):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/query.py: Added.
(session_established):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/server-close.py: Added.
(session_established):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/server-connection-close.py: Added.
(session_established):
(stream_data_received):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.js: Added.
(idl_array.self.webTransport.ready.catch):
(idl_array.self.webTransport.closed.catch):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/in-removed-iframe.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/in-removed-iframe.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/resources/webtransport-test-helpers.sub.js: Added.
(const.BASE.https):
(webtransport_code_to_http_code):
(async read_stream):
(async read_stream_as_string):
(async read_stream_as_json):
(check_and_remove_standard_headers):
(async query):
(async readInto):
(async openWebTransport):
(async openThenCloseWebTransport):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.worker.html: Added.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/webtransport/WebTransport.cpp: Added.
(WebCore::WebTransport::create):
(WebCore::WebTransport::getStats):
(WebCore::createEmptyPromise):
(WebCore::WebTransport::ready):
(WebCore::WebTransport::reliability):
(WebCore::WebTransport::congestionControl):
(WebCore::WebTransport::closed):
(WebCore::WebTransport::draining):
(WebCore::WebTransport::close):
(WebCore::WebTransport::datagrams):
(WebCore::WebTransport::createBidirectionalStream):
(WebCore::WebTransport::incomingBidirectionalStreams):
(WebCore::WebTransport::createUnidirectionalStream):
(WebCore::WebTransport::incomingUnidirectionalStreams):
* Source/WebCore/Modules/webtransport/WebTransport.h: Added.
* Source/WebCore/Modules/webtransport/WebTransport.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportBidirectionalStream.cpp: Added.
(WebCore::WebTransportBidirectionalStream::readable):
(WebCore::WebTransportBidirectionalStream::writable):
* Source/WebCore/Modules/webtransport/WebTransportBidirectionalStream.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportBidirectionalStream.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportCloseInfo.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportCloseInfo.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportCongestionControl.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportCongestionControl.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportDatagramDuplexStream.cpp: Added.
(WebCore::WebTransportDatagramDuplexStream::create):
(WebCore::WebTransportDatagramDuplexStream::readable):
(WebCore::WebTransportDatagramDuplexStream::writable):
(WebCore::WebTransportDatagramDuplexStream::maxDatagramSize):
(WebCore::WebTransportDatagramDuplexStream::incomingMaxAge):
(WebCore::WebTransportDatagramDuplexStream::outgoingMaxAge):
(WebCore::WebTransportDatagramDuplexStream::incomingHighWaterMark):
(WebCore::WebTransportDatagramDuplexStream::outgoingHighWaterMark):
(WebCore::WebTransportDatagramDuplexStream::setIncomingMaxAge):
(WebCore::WebTransportDatagramDuplexStream::setOutgoingMaxAge):
(WebCore::WebTransportDatagramDuplexStream::setIncomingHighWaterMark):
(WebCore::WebTransportDatagramDuplexStream::setOutgoingHighWaterMark):
* Source/WebCore/Modules/webtransport/WebTransportDatagramDuplexStream.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportDatagramDuplexStream.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportDatagramStats.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportDatagramStats.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportError.cpp: Added.
(WebCore::WebTransportError::create):
(WebCore::WebTransportError::WebTransportError):
(WebCore::WebTransportError::source):
(WebCore::WebTransportError::streamErrorCode):
* Source/WebCore/Modules/webtransport/WebTransportError.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportError.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportErrorOptions.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportErrorOptions.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportErrorSource.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportErrorSource.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportHash.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportHash.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportOptions.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportOptions.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportReceiveStream.cpp: Added.
(WebCore::WebTransportReceiveStream::getStats):
* Source/WebCore/Modules/webtransport/WebTransportReceiveStream.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportReceiveStream.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportReceiveStreamStats.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportReceiveStreamStats.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportReliabilityMode.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportReliabilityMode.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStream.cpp: Added.
(WebCore::WebTransportSendStream::getStats):
(WebCore::WebTransportSendStream::sendOrder):
(WebCore::WebTransportSendStream::setSendOrder):
* Source/WebCore/Modules/webtransport/WebTransportSendStream.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStream.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStreamOptions.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStreamOptions.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStreamStats.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStreamStats.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportStats.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportStats.idl: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:

Canonical link: <a href="https://commits.webkit.org/267408@main">https://commits.webkit.org/267408@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/71219c59dad7bd62891e6054c849d67b1c35ea1c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16503 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16825 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17264 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18283 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15477 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16693 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20056 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16968 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/17829 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16700 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/17118 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/14274 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19059 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/14362 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/14958 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/21758 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/14240 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/15350 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/15124 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/19442 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/15740 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15717 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/13348 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/18060 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14918 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/14798 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/4248 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19287 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/19280 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2028 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15543 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/4078 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/DerivedSources.make:
```diff
@@ -768,6 +768,24 @@ JS_BINDING_IDLS := \
     $(WebCore)/Modules/webdriver/Navigator+WebDriver.idl \
     $(WebCore)/Modules/websockets/CloseEvent.idl \
     $(WebCore)/Modules/websockets/WebSocket.idl \
+	$(WebCore)/Modules/webtransport/WebTransport.idl \
```
Indentation issue.

cdumez commented on Source/WebCore/Modules/webtransport/WebTransport.cpp:
```diff
@@ -0,0 +1,109 @@
+}
+
+void WebTransport::getStats(Ref<DeferredPromise>&&)
+{
```
Not a big deal but I usually like to reject promises in stubs. May avoid some time outs in tests.

cdumez commented on Source/WebCore/Modules/webtransport/WebTransportCloseInfo.h:
```diff
@@ -0,0 +1,35 @@
+
+struct WebTransportCloseInfo {
+    unsigned closeCode { 0 };
+    String reason { ""_s };
```
emptyString()

cdumez commented on Source/WebCore/Modules/webtransport/WebTransportError.cpp:
```diff
@@ -0,0 +1,53 @@
+}
+
+WebTransportError::WebTransportError(std::optional<String>&&, std::optional<WebTransportErrorOptions>&&)
+    : DOMException(0, ""_s, ""_s)
```
emptyString(), emptyString()

cdumez commented on Source/WebCore/Modules/webtransport/WebTransportError.h:
```diff
@@ -0,0 +1,43 @@
+
+class WebTransportError : public DOMException {
+public:
+    static Ref<WebTransportError> create(std::optional<String>&&, std::optional<WebTransportErrorOptions>&&);
```
I don't think these 2 parameters should be std::optional<>. They have defaults values in the IDL.

cdumez commented on Source/WebCore/Modules/webtransport/WebTransportReceiveStream.cpp:
```diff
@@ -0,0 +1,35 @@
+namespace WebCore {
+
+void WebTransportReceiveStream::getStats(Ref<DeferredPromise>&&)
+{
```
ditto about rejecting promise.

cdumez commented on Source/WebCore/Modules/webtransport/WebTransportSendStream.cpp:
```diff
@@ -0,0 +1,44 @@
+namespace WebCore {
+
+void WebTransportSendStream::getStats(Ref<DeferredPromise>&&)
+{
```
ditto about rejecting promise.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -7899,6 +7899,19 @@ WebSocketEnabled:
 
+WebTransportEnabled:
+  type: bool
+  status: unstable
```
Note that I think you'll need "testable" for it to get enabled in layout tests iirc.

cdumez reviewed (approved):
In the future, it would be nice to import the tests in a separate (earlier) patch. This is really big.

LGTM with some comments.

achristensen07 committed:
Add IDL skeleton for WebTransport
https://bugs.webkit.org/show_bug.cgi?id=260810
rdar://114578275

Reviewed by Chris Dumez.

This adds IDLs based on the spec at https://www.w3.org/TR/webtransport/
They are off by default and have simple stub implementations that don't do anything yet.
I also imported the web platform tests with a slight modification to webtransport/bidirectional-cancel-crash.https.html
to make it not time out when running with an incomplete implementation.

* LayoutTests/imported/w3c/web-platform-tests/webtransport/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/README.md: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection-ccns.https.tentative.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection-ccns.https.tentative.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection-ccns.https.tentative.window.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-closed-webtransport-connection.https.window.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection-ccns.https.tentative.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection-ccns.https.tentative.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection-ccns.https.tentative.window.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/back-forward-cache-with-open-webtransport-connection.https.window.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/bidirectional-cancel-crash.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/bidirectional-cancel-crash.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/close.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/connect.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.js: Added.
(const.url.of.BAD_URLS.test):
(const.options.of.OPTIONS.promise_test.async t):
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/constructor.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-fail.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-fail.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-fail.https.window.js: Added.
(set_csp):
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-pass.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-pass.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/csp-pass.https.window.js: Added.
(set_csp):
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagram-cancel-crash.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagram-cancel-crash.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagram-cancel-crash.https.window.js: Added.
(test):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.js: Added.
(async write_datagrams):
(async write_N_datagrams):
(async read_datagrams):
(async write_numbers):
(async write_large_datagrams):
(async read_numbers_byob):
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/datagrams.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/echo-large-bidirectional-streams.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/echo-large-bidirectional-streams.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/abort-stream-from-server.py: Added.
(session_established):
(stream_data_received):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/client-close.py: Added.
(session_established):
(stream_reset):
(stream_data_received):
(session_closed):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/custom-response.py: Added.
(connect_received):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/echo-request-headers.py: Added.
(session_established):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/echo.py: Added.
(session_established):
(stream_data_received):
(datagram_received):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/echo_datagram_length.py: Added.
(datagram_received):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/query.py: Added.
(session_established):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/server-close.py: Added.
(session_established):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/handlers/server-connection-close.py: Added.
(session_established):
(stream_data_received):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.js: Added.
(idl_array.self.webTransport.ready.catch):
(idl_array.self.webTransport.closed.catch):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/idlharness.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/in-removed-iframe.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/in-removed-iframe.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/resources/webtransport-test-helpers.sub.js: Added.
(const.BASE.https):
(webtransport_code_to_http_code):
(async read_stream):
(async read_stream_as_string):
(async read_stream_as_json):
(check_and_remove_standard_headers):
(async query):
(async readInto):
(async openWebTransport):
(async openThenCloseWebTransport):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/server-certificate-hashes.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-close.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.js: Added.
(promise_test.async t):
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.serviceworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.serviceworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.sharedworker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.sharedworker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webtransport/streams-echo.https.any.worker.html: Added.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/webtransport/WebTransport.cpp: Added.
(WebCore::WebTransport::create):
(WebCore::WebTransport::getStats):
(WebCore::createEmptyPromise):
(WebCore::WebTransport::ready):
(WebCore::WebTransport::reliability):
(WebCore::WebTransport::congestionControl):
(WebCore::WebTransport::closed):
(WebCore::WebTransport::draining):
(WebCore::WebTransport::close):
(WebCore::WebTransport::datagrams):
(WebCore::WebTransport::createBidirectionalStream):
(WebCore::WebTransport::incomingBidirectionalStreams):
(WebCore::WebTransport::createUnidirectionalStream):
(WebCore::WebTransport::incomingUnidirectionalStreams):
* Source/WebCore/Modules/webtransport/WebTransport.h: Added.
* Source/WebCore/Modules/webtransport/WebTransport.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportBidirectionalStream.cpp: Added.
(WebCore::WebTransportBidirectionalStream::readable):
(WebCore::WebTransportBidirectionalStream::writable):
* Source/WebCore/Modules/webtransport/WebTransportBidirectionalStream.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportBidirectionalStream.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportCloseInfo.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportCloseInfo.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportCongestionControl.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportCongestionControl.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportDatagramDuplexStream.cpp: Added.
(WebCore::WebTransportDatagramDuplexStream::create):
(WebCore::WebTransportDatagramDuplexStream::readable):
(WebCore::WebTransportDatagramDuplexStream::writable):
(WebCore::WebTransportDatagramDuplexStream::maxDatagramSize):
(WebCore::WebTransportDatagramDuplexStream::incomingMaxAge):
(WebCore::WebTransportDatagramDuplexStream::outgoingMaxAge):
(WebCore::WebTransportDatagramDuplexStream::incomingHighWaterMark):
(WebCore::WebTransportDatagramDuplexStream::outgoingHighWaterMark):
(WebCore::WebTransportDatagramDuplexStream::setIncomingMaxAge):
(WebCore::WebTransportDatagramDuplexStream::setOutgoingMaxAge):
(WebCore::WebTransportDatagramDuplexStream::setIncomingHighWaterMark):
(WebCore::WebTransportDatagramDuplexStream::setOutgoingHighWaterMark):
* Source/WebCore/Modules/webtransport/WebTransportDatagramDuplexStream.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportDatagramDuplexStream.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportDatagramStats.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportDatagramStats.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportError.cpp: Added.
(WebCore::WebTransportError::create):
(WebCore::WebTransportError::WebTransportError):
(WebCore::WebTransportError::source):
(WebCore::WebTransportError::streamErrorCode):
* Source/WebCore/Modules/webtransport/WebTransportError.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportError.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportErrorOptions.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportErrorOptions.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportErrorSource.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportErrorSource.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportHash.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportHash.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportOptions.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportOptions.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportReceiveStream.cpp: Added.
(WebCore::WebTransportReceiveStream::getStats):
* Source/WebCore/Modules/webtransport/WebTransportReceiveStream.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportReceiveStream.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportReceiveStreamStats.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportReceiveStreamStats.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportReliabilityMode.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportReliabilityMode.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStream.cpp: Added.
(WebCore::WebTransportSendStream::getStats):
(WebCore::WebTransportSendStream::sendOrder):
(WebCore::WebTransportSendStream::setSendOrder):
* Source/WebCore/Modules/webtransport/WebTransportSendStream.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStream.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStreamOptions.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStreamOptions.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStreamStats.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportSendStreamStats.idl: Added.
* Source/WebCore/Modules/webtransport/WebTransportStats.h: Added.
* Source/WebCore/Modules/webtransport/WebTransportStats.idl: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:

Canonical link: https://commits.webkit.org/267408@main
